modules/client/main.tf

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#--------#
+# Locals #
+#--------#
+locals {
+  client_startup_script = "${file("${path.module}/templates/scripts/forseti-client/forseti_client_startup_script.sh.tpl")}"
+  client_env_script     = "${file("${path.module}/templates/scripts/forseti-client/forseti_environment.sh.tpl")}"
+  client_conf           = "${file("${path.module}/templates/configs/forseti_conf_client.yaml.tpl")}"
+  client_conf_path      = "${var.forseti_home}/configs/forseti_conf_client.yaml"
+  client_sa_name        = "forseti-client-gcp-${var.suffix}"
+  client_name           = "forseti-client-vm-${var.suffix}"
+  client_bucket_name    = "forseti-client-${var.suffix}"
+  client_zone           = "${var.client_region}-c"
+
+  client_project_roles = [
+    "roles/storage.objectViewer",
+    "roles/cloudtrace.agent",
+  ]
+}
+
+#-------------------#
+# Forseti templates #
+#-------------------#
+data "template_file" "forseti_client_startup_script" {
+  template = "${local.client_startup_script}"
+
+  vars {
+    forseti_environment      = "${data.template_file.forseti_client_environment.rendered}"
+    forseti_repo_url         = "${var.forseti_repo_url}"
+    forseti_version          = "${var.forseti_version}"
+    forseti_home             = "${var.forseti_home}"
+    forseti_client_conf_path = "${local.client_conf_path}"
+    storage_bucket_name      = "${local.client_bucket_name}"
+  }
+}
+
+data "template_file" "forseti_client_environment" {
+  template = "${local.client_env_script}"
+
+  vars {
+    forseti_home             = "${var.forseti_home}"
+    forseti_client_conf_path = "${local.client_conf_path}"
+  }
+}
+
+data "template_file" "forseti_client_config" {
+  template = "${local.client_conf}"
+
+  vars {
+    forseti_server_ip = "${var.server_address}"
+  }
+}
+
```

Comment:
Current firewall rules will not allow for ssh connection to forseti-client. Forseti-server module firewall rules will block all ssh traffic to forseti-client.

Reply:
@ocervell and I discussed this via chat, and we didn't see a need to permit SSH from the Forseti server to the Forseti client. If a use case does come up we can address it later on.